assets/js/atomic/blocks/product-elements/add-to-cart/product-types/variable/variation-attributes/attribute-select-control.js

```diff
@@ -19,17 +23,67 @@ const selectAnOption = {
 const AttributeSelectControl = ( {
 	attributeName,
 	options = [],
-	selected = '',
+	value = '',
 	onChange = () => {},
+	errorMessage = __(
+		'Please select a value.',
+		'woo-gutenberg-products-block'
+	),
 } ) => {
+	const {
+		getValidationError,
+		setValidationErrors,
+		clearValidationError,
+	} = useValidationContext();
+	const errorId = attributeName;
+	const error = getValidationError( errorId ) || {};
+
+	useEffect( () => {
+		if ( value ) {
+			clearValidationError( errorId );
+		} else {
+			setValidationErrors( {
+				[ errorId ]: {
+					message: errorMessage,
+					hidden: true,
+				},
+			} );
+		}
+	}, [
+		value,
+		errorId,
+		errorMessage,
+		clearValidationError,
+		setValidationErrors,
+	] );
+
+	// Remove validation errors when unmounted.
+	useEffect( () => {
+		return () => {
+			clearValidationError( errorId );
+		};
+	}, [ errorId, clearValidationError ] );
+
 	return (
-		<SelectControl
-			className="wc-block-components-product-add-to-cart-attribute-picker__select"
-			label={ decodeEntities( attributeName ) }
-			value={ selected || '' }
-			options={ [ selectAnOption, ...options ] }
-			onChange={ onChange }
-		/>
+		<div className="wc-block-components-product-add-to-cart-attribute-picker__content">
+			<SelectControl
+				label={ decodeEntities( attributeName ) }
+				value={ value || '' }
+				options={ [ selectAnOption, ...options ] }
+				onChange={ onChange }
+				required={ true }
+				className={ classnames(
+					'wc-block-components-product-add-to-cart-attribute-picker__select',
+					{
+						'has-error': error.message && ! error.hidden,
+					}
+				) }
+			/>
+			<ValidationInputError
```

Comment:
It was not possible to reuse the checkout selects because they use the CustomSelect component and our designs called for native controls.

Reply:
The idea is to not custom style the selects. This was a design decision. Personally I wanted to use checkout style selects.